website/server/libs/cron.js

```diff
@@ -381,7 +383,19 @@ export function cron (options = {}) {
       });
     }
 
-    task.completed = false;
+    // If this is a shared task, check if another user completed it in the "same" day the user is "starting"
+    if (task.group && task.group.sharedCompletion === SHARED_COMPLETION.single) {
+      // @REVIEW This introduces an async call into this cron function.
+      // The function is called from ../middlewares/cron.js asyncCron(), which is async and suggests this is okay
+      // Does this cause issues?
+      // Pushing these closures to an array of Promises outside the forEach
+      // This allows both sequential processing and async checking for shared completion
+      groupSharedSingleDailies.push(async function determineGroupCompletion (memberTask, memberUser, memberTime) {
+        memberTask.completed = await groupTaskCompleted(memberTask, memberUser, memberTime);
+      }(task, user, now));
```

Comment:
The function this is inside (`cron (options = {})`) was synchronous. It was called from the asynchronous `cronAsync()` in the `website/server/middlewares/cron.js` file.

1. I do not know the ramifications of `await`ing this cron call from the middleware. On the surface, it should only increase the time to process, and only if the user has Single Completion Shared Dailies.
2. I tried to make this pattern of pushing an executed asynchronous closure to an array `await`ed outside the `forEach` loop more elegant with an arrow function, but it never executed. If there are suggestions about making this more elegant in any way, please let me know. I wanted to keep the sequential `forEach` for two reasons:
    * It keeps the pattern used to iterate the other task types
    * It keeps the logic sequential. Since there are variables outside the loop scope that are modified and checked, this will avoid tricky issues now and in the future.